src/System.Private.CoreLib/src/System/Threading/Timer.Unix.cs

```diff
@@ -13,109 +14,128 @@ namespace System.Threading
     //
     // Unix-specific implementation of Timer
     //
-    internal partial class TimerQueue
+    internal partial class TimerQueue : IThreadPoolWorkItem
     {
+        private static readonly Lock s_lock = new Lock();
+        private static List<AppDomainTimer> s_appDomainTimers;
+
         /// <summary>
         /// This event is used by the timer thread to wait for timer expiration. It is also
         /// used to notify the timer thread that a new timer has been set.
         /// </summary>
-        private static AutoResetEvent s_timerEvent;
+        private static readonly AutoResetEvent s_timerEvent = new AutoResetEvent(false);
 
-        /// <summary>
-        /// This field stores the value of next timer that the timer thread should install.
-        /// </summary>
-        private static volatile int s_nextTimerDuration;
+        private readonly int _id;
 
         private TimerQueue(int id)
         {
+            _id = id;
+        }
+
+        private static List<AppDomainTimer> InitializeAppDomainTimerManager_Locked()
```

Comment:
I assume you took the "AppDomain" part of the name from CoreCLR. I'd rather refrain from using that. I was encouraged to remove it on the CoreCLR side and introducing it in a new code is kinda going backwards. Would something like `ScheduledTimer` work here?